### ISSUE TITLE
Big-endian builds

### ISSUE THREAD

**Issue by joeshaw** [chitoku-k/edgerouter-exporter#143]
I have an EdgeRouter 12 which is a big-endian mips64 machine, and the provided binaries of edgerouter-exporter are little-endian.  Would you consider building big-endian binaries as well?  Doesn't matter to me if they are 32- or 64-bit.

```
$ file edgerouter-exporter
edgerouter-exporter: ELF 32-bit LSB pie executable, MIPS, MIPS32 rel2 version 1 (SYSV), dynamically linked, interpreter /lib/ld.so.1, for GNU/Linux 3.2.0, BuildID[sha1]=87f795e3315b341a73f495423a344ab828922f4d, stripped
```

A random binary from `/usr/bin` on my ER12:

```
$ file uuidgen
uuidgen: ELF 32-bit MSB pie executable, MIPS, MIPS32 rel2 version 1 (SYSV), dynamically linked, interpreter /lib/ld.so.1, for GNU/Linux 3.2.0, BuildID[sha1]=78aee1065bd065be32125f3f81342258a3abea94, stripped
```

And tailscale, of which I download the mips64 version:

```
tailscale: ELF 64-bit MSB executable, MIPS, MIPS-III version 1 (SYSV), statically linked, Go BuildID=n-W1VBy_aY7zL5CMsfla/DvgkkUAzZvxN07tiM8B1/mGCincyflrToFX0-o7sz/ievfa6WmUWpBHKO-nlOI, not stripped
```

**Comment by chitoku-k** [comment]
Currently, edgerouter-exporter has not provided big-endian builds just because I'm not sure whether it works on Cavium-based devices. I will cut a release with a mips64 binary as well in the next release in the hope that you would try it out. Thanks!

**Comment by joeshaw** [comment]
I appreciate you doing the build.  Unfortunately it didn't work for me:

```
joe@router:~$ ./edgerouter-exporter-mips64
-vbash: ./edgerouter-exporter-mips64: Accessing a corrupted shared library
joe@router:~$ ldd ./edgerouter-exporter-mips64
	not a dynamic executable
```

I got the same when I built from source myself.  I think we need to be targeting `mips64-unknown-linux-muslabi64` instead.  When I build with that the program starts but dies with this error:

```
joe@router:~$ PORT=9999 ./edgerouter-exporter
thread 'main' panicked at 'failed to create UnixStream: Os { code: 22, kind: InvalidInput, message: "Invalid argument" }', /cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.19.2/src/signal/unix.rs:60:53
```

I haven't tried debugging it further than that yet.

**Comment by chitoku-k** [comment]
Thanks for trying this out! From the error message, I guess the system has only 32-bit shared libraries if I understand this correctly. Either targeting `mips64-unknown-linux-muslabi64` or `mips-unknown-linux-gnu` should also work too.

For the `Invalid argument`, make sure you have the right permission to access `/run/charon.vici`. It looks like you are running edgerouter-exporter as a regular user from your log, which is not supported.

**Comment by joeshaw** [comment]
On the 64-bit binary, I don't think it's related to `/run/charon.vici` because it doesn't matter if I run it as root or not, I always get that.

The 32-bit binary built using the target you suggested works fine.

**Comment by chitoku-k** [comment]
> The 32-bit binary built using the target you suggested works fine.

OK. I'll change the build target to `mips-unknown-linux-gnu` from the next release then. I guess some internal initialization for UNIX sockets might not be working when using musl libc.

**Comment by chitoku-k** [comment]
Closing as it should be solved in v2.5.1. Feel free to reopen if any.
